### PR TITLE
Fix linter warnings, part 1

### PR DESCRIPTION
**Pull request by jefferai**
This is part 1 of some work to fix linter warnings, cut off just to keep the PR size reasonable.

### CODE REVIEW

File: internal/credential/vault/jobs.go [hunk 1]
```diff
@@ -270,7 +270,7 @@ func nextRenewal(ctx context.Context, j scheduler.Job) (time.Duration, error) {
 	}
 	defer rows.Close()
 
-	for rows.Next() {
+	if rows.Next() {
```
Comment (johanbrandhorst) [inline internal/credential/vault/jobs.go:273]: Is this change correct? I am aware that `r.ScanRows` consumes the rows, but isn't it still safer to use a `for` loop?

File: internal/event/options_test.go [hunk 2]
```diff
@@ -5,8 +5,8 @@ package event
 
 import (
 	"context"
+	"crypto/rand"
```
Comment (johanbrandhorst) [inline internal/event/options_test.go:8]: 😰

File: internal/server/root_certificate.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ func newRootCertificate(ctx context.Context, serialNumber uint64, certificate []
 ) (*RootCertificate, error) {
 	const op = "server.newRootCertificate"
 
-	if &serialNumber == nil {
```
Comment (johanbrandhorst) [inline internal/server/root_certificate.go:92]: 😰

**Review by johanbrandhorst — APPROVED** [review]
👏🏻 awesome!!

File: internal/credential/vault/jobs.go (hunk 1 above)
Comment (jefferai) [reply internal/credential/vault/jobs.go:273]: I checked with @louisruch and he confirmed that we should only ever want one row (or zero). The linter had been complaining about the fact that there was an unconditional return in the for loop.

If we need to ScanRows to consume all of them or something we can... @louisruch thoughts here?

File: internal/event/options_test.go (hunk 2 above)
Comment (jefferai) [reply internal/event/options_test.go:8]: It's fine, it was seeded anyways :-D

File: internal/server/root_certificate.go (hunk 3 above)
Comment (jefferai) [reply internal/server/root_certificate.go:92]: :-D

File: internal/credential/vault/jobs.go (hunk 1 above)
Comment (johanbrandhorst) [reply internal/credential/vault/jobs.go:273]: Oh, I see. Yeah I mean I'd probably change the return to a break?

Comment (louisruch) [reply internal/credential/vault/jobs.go:273]: This looks good to me, a break would also work

**Review by louisruch — APPROVED** [review]
this looks great!

File: internal/credential/vault/jobs.go (hunk 1 above)
Comment (jefferai) [reply internal/credential/vault/jobs.go:273]: Adding a break suggests that maybe if something goes wrong with the first row that we want to process a second row? But @louisruch said we should really expect only one, which the `if` will guarantee. So personally I'm leaning towards how it is now if that's okay with you two.

Comment (louisruch) [reply internal/credential/vault/jobs.go:273]: `if` sounds good to me

**Pull request merged**